.github/workflows/typecheck.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  workflow_call:
+
+jobs:
+  tsc:
+    name: Check
+    runs-on: ubuntu-latest
+    timeout-minutes: 4
+    defaults:
+      run:
+        working-directory: web
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7
+      - name: Setup Node v18.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Restore node_modules for web
+        id: cache-web
+        uses: actions/cache@v3
+        with:
+          path: ./web/node_modules
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('./web/pnpm-lock.yaml') }}
+      - name: Install web dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: pnpm install --frozen-lockfile
+      - name: Run TypeScript typechecking
+        run: pnpm run run-tsc
```

Comment:
hehe we should probably change the names of these commands, run run-tsc looks weird 😅

But we can do that in a separate PR.

Reply:
Haha yeah, it is a bit weird :D